integration/util.py

```diff
@@ -835,18 +901,20 @@ def await_file_contents(path, contents, timeout=15):
             Message.log(
                 message_type=u"integration:await-file-contents:missing",
             )
-        time.sleep(1)
+        yield twisted_sleep(reactor, 1)
     if exists(path):
         raise ExpectedFileMismatchException(path, timeout)
     raise ExpectedFileUnfoundException(path, timeout)
 
+@inline_callbacks
 def ensure_file_not_created(path, timeout=15):
+    from twisted.internet import reactor
     start_time = time.time()
     while time.time() - start_time < timeout:
```

Comment:
It probably doesn't make any difference, at least not now, but as long as this is reactor-driven code `reactor.seconds` might make more sense than `time.time` here?